api/configuration/v1beta1/kongconsumergroup_types.go

```diff
@@ -28,20 +31,75 @@ import (
 // +kubebuilder:resource:shortName=kcg,categories=kong-ingress-controller
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`,description="Age"
 // +kubebuilder:printcolumn:name="Programmed",type=string,JSONPath=`.status.conditions[?(@.type=="Programmed")].status`
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.spec.controlPlaneRef) || has(self.spec.controlPlaneRef)", message="controlPlaneRef is required once set"
+// +kubebuilder:validation:XValidation:rule="(!has(self.status) || !self.status.conditions.exists(c, c.type == 'Programmed' && c.status == 'True')) ? true : oldSelf.spec.controlPlaneRef == self.spec.controlPlaneRef", message="spec.controlPlaneRef is immutable when entity is already Programmed."
 
 // KongConsumerGroup is the Schema for the kongconsumergroups API.
 type KongConsumerGroup struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	Spec KongConsumerGroupSpec `json:"spec,omitempty"`
+
 	// Status represents the current status of the KongConsumerGroup resource.
 	Status KongConsumerGroupStatus `json:"status,omitempty"`
 }
 
+type KongConsumerGroupSpec struct {
+	// Name is the name of the ConsumerGroup in Kong.
+	Name *string `json:"name,omitempty"`
+
+	// ControlPlaneRef is a reference to a ControlPlane this ConsumerGroup is associated with.
+	// +optional
+	ControlPlaneRef *configurationv1alpha1.ControlPlaneRef `json:"controlPlaneRef,omitempty"`
+}
+
+func (c *KongConsumerGroup) InitKonnectStatus() {
```

Comment:
How about keeping this private? It's used in setters, but I don't think we should leak this outside (the same applies to KongConsumer, KongRoute and KongService).